Rendering/VolumeOpenGL2/vtkVolumeShaderComposer.h

```diff
@@ -1258,6 +1259,8 @@ inline std::string ComputeLightingDeclaration(vtkRenderer* vtkNotUsed(ren), vtkV
 
   int const transferMode = volProperty->GetTransferFunctionMode();
 
+  std::string lightingComponent = independentComponents ? "component" : "0";
+
```

Comment:
After review the implementation, I suggest we remove the use of `regex` and instead add a `#define`.

1. Append `#define` ...

```
if (independentComponents)
{
  shaderStr += "#define LIGHTING_COMPONENT component"
}
else
{
  shaderStr += "#define LIGHTING_COMPONENT 0"
}
```

2. Use `LIGHTING_COMPONENT` instead of `__LIGHTING_COMPONENT__`

cc: @lassoan @pieper

Reply:
I'm OK with it. Ultimately VTK renderer developers will need to approve it anyway.
(I would consider a simple variable to be safer, as its scope would be more controlled than of a preprocessor macro)